### PR TITLE
Feature/fault injection

### PR DESCRIPTION
**Pull request by PiotrMrozik**

### CODE REVIEW

File: src/gpu/RaytraceRequestContext.hpp [hunk 1]
```diff
@@ -41,6 +41,8 @@ struct RaytraceRequestContext
 	const float* rayTimeOffsets;
 	size_t rayTimeOffsetsCount;
 
+	const int* rayMask;
```
Comment (prybicki) [inline src/gpu/RaytraceRequestContext.hpp:44]: Minor: please use types consistently, `int` is not necessarily `int32_t`

File: src/graph/NodesCore.hpp [hunk 2]
```diff
@@ -126,6 +126,7 @@ struct RaytraceNode : IPointsNode
 	void setVelocity(const Vec3f& linearVelocity, const Vec3f& angularVelocity);
 	void enableRayDistortion(bool enabled) { doApplyDistortion = enabled; }
 	void setNonHitDistanceValues(float nearDistance, float farDistance);
+	void setNonHitsMask(const int* maskRaw, size_t maskPointCount);
```
Comment (prybicki) [inline src/graph/NodesCore.hpp:129]: ditto: type consistency

File: src/graph/RaytraceNode.cpp [hunk 3]
```diff
@@ -94,6 +99,7 @@ void RaytraceNode::enqueueExecImpl()
 	    .ringIdsCount = ringIds.has_value() ? (*ringIds)->getCount() : 0,
 	    .rayTimeOffsets = timeOffsets.has_value() ? (*timeOffsets)->asSubclass<DeviceAsyncArray>()->getReadPtr() : nullptr,
 	    .rayTimeOffsetsCount = timeOffsets.has_value() ? (*timeOffsets)->getCount() : 0,
+	    .rayMask = rayMask.has_value() ? (*rayMask)->getReadPtr() : nullptr,
```
Comment (prybicki) [inline src/graph/RaytraceNode.cpp:102]: Ditto simplify

File: src/graph/RaytraceNode.cpp [hunk 4]
```diff
@@ -177,3 +183,8 @@ void RaytraceNode::setNonHitDistanceValues(float nearDistance, float farDistance
 	nearNonHitDistance = nearDistance;
 	farNonHitDistance = farDistance;
 }
+void RaytraceNode::setNonHitsMask(const int* maskRaw, size_t maskPointCount)
+{
+	(*rayMask)->copyFromExternal(maskRaw, maskPointCount);
+	maskCount = maskPointCount;
```
Comment (prybicki) [inline src/graph/RaytraceNode.cpp:189]: Ditto: simplify

File: test/src/graph/MaskRaysTest.cpp [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+	TestPointCloud outputPointCloud = TestPointCloud::createFromNode(yieldNode, fields);
+	auto fullCloudSize = outputPointCloud.getPointCount();
+
+	EXPECT_RGL_SUCCESS(rgl_node_raytrace_configure_mask(raytraceNode, points_mask.data(), raysNumber));
+
+	EXPECT_RGL_SUCCESS(rgl_graph_run(raytraceNode));
+	TestPointCloud outputPointCloudMasked = TestPointCloud::createFromNode(yieldNode, fields);
+	auto maskedCloudSize = outputPointCloudMasked.getPointCount();
+
+	EXPECT_EQ(fullCloudSize - maskCount, maskedCloudSize);
```
Comment (prybicki) [inline test/src/graph/MaskRaysTest.cpp:96]: You can also easily verify if the proper rays are hidden using RAY_IDX field.

**Review by prybicki — CHANGES_REQUESTED** [review]

File: test/src/graph/MaskRaysTest.cpp (hunk 5 above)
Comment (PiotrMrozik) [reply test/src/graph/MaskRaysTest.cpp:96]: I can.
Not rly sure if I want in this test.

**Review by prybicki — APPROVED** [review]

**Pull request merged**